test/integration/dynamic-routing/test/index.test.js

```diff
@@ -156,15 +156,18 @@ function runTests () {
     const html = await renderViaHTTP(appPort, '/on-mount/post-1')
     expect(html).not.toMatch(/post:.*?\[post\].*?<\/p>/)
   })
+
+  it('should update with a hash in the URL', async () => {
+    const browser = await webdriver(appPort, '/on-mount/post-1#abc')
```

Comment:
It might be good to also test client transitions for dynamic routes with a hash

Reply:
Added